### PR TITLE
Add more problem types to instructor queries

### PR DESCRIPTION
**Pull request by sigberto**
Now included are: submit and compare, free text response, inline dropdown, MUFI, image coding,
peer assessment, qualtrics survey and poll

### CODE REVIEW

File: lms/djangoapps/instructor/views/data_access_constants.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ class Inclusion:
                          'F': Inclusion.FILTER}
 
 INCLUDE_SECTION_PATTERN = re.compile('chapter|sequential')
-INCLUDE_PROBLEM_PATTERN = re.compile('problem')
+INCLUDE_PROBLEM_PATTERN = re.compile('problem|openassessment|submit-and-compare|freetextresponse|xblockmufi|qualtricssurvey|inline-dropdown|poll|image-coding')
```
Comment (stvstnfrd) [inline lms/djangoapps/instructor/views/data_access_constants.py]: I wonder if it would be easier to read/maintain if we broke it up, something like:

```python
INCLUDE_PROBLEMS = [
    'problem',
    'openassessment',
    'submit-and-compare',
    # etc...
]
INCLUDE_PROBLEM_PATTERN = re.compile('|'.join(INCLUDE_PROBLEMS))
```

Comment (dcadams) [inline lms/djangoapps/instructor/views/data_access_constants.py]: Even better, have this list as a setting since it is likely to change often.

Comment (dcadams) [inline lms/djangoapps/instructor/views/data_access_constants.py]: As an example, look for INLINE_ANALYTICS_SUPPORTED_TYPES in lms.env.common and lms.env.aws and follow a similar pattern.

File: lms/djangoapps/instructor/views/data_access_constants.py [hunk 2]
```diff
@@ -48,8 +49,8 @@ class Inclusion:
                          'N': Inclusion.NOT,
                          'F': Inclusion.FILTER}
 
-INCLUDE_SECTION_PATTERN = re.compile('chapter|sequential')
-INCLUDE_PROBLEM_PATTERN = re.compile('problem')
+INCLUDE_SECTION_PATTERN = re.compile('|'.join(settings.INSTRUCTOR_QUERY_SECTION_TYPES))
```
Comment (stvstnfrd) [inline lms/djangoapps/instructor/views/data_access_constants.py]: Let's not future-proof this just yet, which is to say,
let's undo this for now.

I'm not sure what else this would be; @dcadams ?

File: lms/envs/common.py [hunk 3]
```diff
@@ -2870,3 +2870,14 @@
 TYPES_WITH_CHILD_PROBLEMS_LIST = [
     'library_content',
 ]
+
+# These are the problem types (by block_type) that an instructor can query over in the Instructor Query tab
+INSTRUCTOR_QUERY_PROBLEM_TYPES = [
```
Comment (stvstnfrd) [inline lms/envs/common.py]: You'll still need the corresponding entry in `aws.py`,
else your configuration/json will never be read in ;)

```python
INSTRUCTOR_QUERY_PROBLEM_TYPES = ENV_TOKENS.get('INSTRUCTOR_QUERY_PROBLEM_TYPES', INSTRUCTOR_QUERY_PROBLEM_TYPES)
```

**Comment by stvstnfrd** [comment]
Also, take a look at the test failure; I believe that was introduced here:

https://jenkins.class.stanford.edu:8080/job/Stanford_Online_Unit_Tests_auto_pr/1716/console

File: lms/djangoapps/instructor/views/data_access_constants.py (hunk 2 above)
Comment (dcadams) [inline lms/djangoapps/instructor/views/data_access_constants.py]: Any xblock that Monica didn't think of at the time or any new one. There's a precedent for putting these types of lists in common (STUDENT_RESPONSES_REPORT_SUPPORTED_TYPE, INLINE_ANALYTICS_SUPPORTED_TYPES). Why would we not want to put this in common @stvstnfrd ?

Comment (stvstnfrd) [inline lms/djangoapps/instructor/views/data_access_constants.py]: @dcadams @sigberto Sorry, my comment was referencing the `chapter|sequential` bit.
I agree problem types should be customized.

@dcadams Grab me in person if we're still on a different page.

File: lms/envs/common.py (hunk 3 above)
Comment (sigberto) [inline lms/envs/common.py]: @stvstnfrd is there a particular place where we insert this kind of thing? or is the list of vars in aws.py more or less random?

File: lms/djangoapps/instructor/views/data_access_constants.py (hunk 2 above)
Comment (sigberto) [inline lms/djangoapps/instructor/views/data_access_constants.py]: @stvstnfrd so, just to be clear, we're only changing back the `chapter|sequential` bit, right?

File: lms/envs/common.py (hunk 3 above)
Comment (stvstnfrd) [inline lms/envs/common.py]: @sigberto We're working on standardizing that, but in the interim, just stick it on the bottom, I guess? Or better yet, maybe near `TYPES_WITH_CHILD_PROBLEMS_LIST` if it's there? (or something similar)?

Comment (stvstnfrd) [inline lms/envs/common.py]: @sigberto Correct. I do not foresee us overriding `INCLUDE_SECTION_PATTERN` at this time.

**Comment by sigberto** [comment]
@dcadams technically yes, but when you try to run cms without the variables in cms/.../[common|aws].py, paver crashes. i think it's because the data_access_constants.py file is loaded for both cms and lms

**Comment by dcadams** [comment]
@sig, I dug into this and yes, it percolates into common (common/djangoapps/student/views). I don't see a better way than what you are doing.

File: lms/envs/common.py [hunk 4]
```diff
@@ -2870,3 +2870,8 @@
...
+    'problem'
+]
```
Comment (dcadams) [inline lms/envs/common.py]: Aren't you missing that list of xblocks here?

Comment (sigberto) [inline lms/envs/common.py]: @dcadams that's what will go in the playbook

File: cms/envs/common.py [hunk 5]
```diff
@@ -47,6 +47,8 @@
 
     # django-debug-toolbar
     DEBUG_TOOLBAR_PATCH_SETTINGS,
+
+    INSTRUCTOR_QUERY_PROBLEM_TYPES,
```
Comment (stvstnfrd) [inline cms/envs/common.py]: Oh, never mind; I saw your comments elsewhere here.

Comment (sigberto) [inline cms/envs/common.py]: it was at the very start of running paver (within seconds). i think the error log mentioned configuration, but I didn't write it down. could check if you're curious

File: lms/envs/common.py (hunk 4 above)
Comment (dcadams) [inline lms/envs/common.py]: Why isn't 'problem' going into the playbook?

Comment (sigberto) [inline lms/envs/common.py]: because that's the default. it's what edx had

Comment (dcadams) [inline lms/envs/common.py]: It's actually what JimC had not edx. In my mind, this makes it really confusing. If I'm trying to debug an issue and I want to see what types are supported I have to remember that this list is being overridden by another list in another repo. It would be easier to figure that out if this list was empty. Just my 2p.

File: cms/envs/aws.py [hunk 6]
```diff
@@ -364,3 +364,5 @@
 
 PROCTORING_BACKEND_PROVIDER = AUTH_TOKENS.get("PROCTORING_BACKEND_PROVIDER", PROCTORING_BACKEND_PROVIDER)
 PROCTORING_SETTINGS = ENV_TOKENS.get("PROCTORING_SETTINGS", PROCTORING_SETTINGS)
+
+INSTRUCTOR_QUERY_PROBLEM_TYPES = ENV_TOKENS.get('INSTRUCTOR_QUERY_PROBLEM_TYPES', INSTRUCTOR_QUERY_PROBLEM_TYPES)
```
Comment (stvstnfrd) [inline cms/envs/aws.py]: Is this a duplicate of `openedx/stanford/lms/envs/aws.py`?

File: cms/envs/common.py (hunk 5 above)
Comment (stvstnfrd) [inline cms/envs/common.py]: I think this can be moved to `openedx/stanford/cms/envs/common.py`

File: lms/envs/aws.py [hunk 7]
```diff
@@ -731,4 +731,4 @@
 ################# PROCTORING CONFIGURATION ##################
 
 PROCTORING_BACKEND_PROVIDER = AUTH_TOKENS.get("PROCTORING_BACKEND_PROVIDER", PROCTORING_BACKEND_PROVIDER)
-PROCTORING_SETTINGS = ENV_TOKENS.get("PROCTORING_SETTINGS", PROCTORING_SETTINGS)
+PROCTORING_SETTINGS = ENV_TOKENS.get("PROCTORING_SETTINGS", PROCTORING_SETTINGS)
```
Comment (stvstnfrd) [inline lms/envs/aws.py]: This is unintentional (you're removing the trailing newline);
please undo this change.

File: lms/envs/common.py [hunk 8]
```diff
@@ -2742,4 +2742,4 @@
 # This is an arbitrary hard limit.
 # The reason we introcuced this number is because we do not want the CCX
 # to compete with the MOOC.
-CCX_MAX_STUDENTS_ALLOWED = 200
+CCX_MAX_STUDENTS_ALLOWED = 200
```
Comment (stvstnfrd) [inline lms/envs/common.py]: Same here. I suspect your text editor is incorrectly configured and is stripping the trailing newlines.

File: openedx/stanford/lms/envs/common.py [hunk 9]
```diff
@@ -92,3 +92,8 @@
     # These types are children of children of units.
     'library_content',
 ]
+
+# These are the problem types (by block_type) that an instructor can query over in the Instructor Query tab
+INSTRUCTOR_QUERY_PROBLEM_TYPES = [
+    'problem'
```
Comment (stvstnfrd) [inline openedx/stanford/lms/envs/common.py]: Add a trailing comma.

File: openedx/stanford/lms/envs/aws.py [hunk 10]
```diff
@@ -35,3 +35,5 @@
     'STUDENT_RESPONSES_DOWNLOAD',
     STUDENT_RESPONSES_DOWNLOAD
 )
+
+INSTRUCTOR_QUERY_PROBLEM_TYPES = ENV_TOKENS.get('INSTRUCTOR_QUERY_PROBLEM_TYPES', INSTRUCTOR_QUERY_PROBLEM_TYPES)
```
Comment (stvstnfrd) [inline openedx/stanford/lms/envs/aws.py]: This is an awfully long line; can we split it following the form above?

**Comment by sigberto** [comment]
@stvstnfrd made all the changes

**Comment by sigberto** [comment]
@stvstnfrd is the build still running? or why have there been no updates on this?

**Comment by stvstnfrd** [comment]
@sigberto I did trigger a rebuild, not sure why it's not syncing up. Can you try force-pushing?
